client/src/pages/Search/Search.tsx

```diff
@@ -1,3 +1,92 @@
+import ResultCard from './ResultCard/ResultCard';
+import DrawerFilter from './Filter/DrawerFilter';
+import AsideFilter from './Filter/AsideFilter';
+import { Link, useSearchParams } from 'react-router-dom';
+import { useEffect } from 'react';
+import { Button } from '@/components/ui/button';
+import useSWRInfinite, { SWRInfiniteKeyLoader } from 'swr/infinite';
+import { SearchData } from '@/types/api-types';
+import { AxiosRequestConfig } from 'axios';
+import { toPreview } from '@/constants/routes';
+
 export default function Search() {
-  return <div>Search</div>;
+  const [searchParams, _] = useSearchParams();
+
+  const getRequestObj = (page: number) => {
+    const params = {} as Record<string, string | string[]>;
+    searchParams.forEach((value, key) => {
+      if (key === 'year') {
+        params[key] = value.split(',').map((item) => item.trim());
+      } else if (key === 'query') {
+        params['searchParams'] = value.split(',').map((item) => item.trim());
+      } else if (key) params[key] = value;
+    });
+
+    params['page'] = String(page);
+
+    const requestObj: AxiosRequestConfig<any> = {
+      url: '/search?',
+      params,
+      method: 'GET',
+    };
+
+    return requestObj;
+  };
+
+  const getKey: SWRInfiniteKeyLoader = (pageIndex, previousPageData) => {
+    // TODO: Content of previousPageData needs further testing
+
+    if (previousPageData && !previousPageData.hasMore) return null;
+    return getRequestObj(pageIndex + 1);
+  };
+
+  const {
+    data: response,
+    setSize,
```

Comment:
Okay this implementation looks fine. But IMO its bit complicated. We can simply define a state in a parent component and pass the state to its child components and to update the state we can pass the event handlers to the child component. The only communication that would happen is btw the state defined in parent component and the URL. IMO that implementation looks more clean.

Ref [this gist](https://gist.github.com/Bismay5467/854fedfb2ada3caf6ce5fe8d3fb6101a) for any confusion

PS: this review includes the code defined in `FilterForm` component.